SnP_Emulator.py

```diff
@@ -0,0 +1,273 @@
+        rec_sip_deadband = avg_ambient_pressure - max(pressures['soft_sip'])
+
+        rec_deadband_size = min(rec_puff_deadband, rec_sip_deadband)
+
+        print("Recommended values:")
+        print("\t AMBIENT = {}".format(self._ambient_pressure))
+        print("\t PUFF: threshold = {}, deadband = {}".format(rec_puff_threshold, rec_puff_deadband))
+        print("\t SIP: threshold = {}, deadband = {}".format(rec_sip_threshold, rec_sip_deadband))
+        print("\t Overall Deadband: ", rec_deadband_size)
+
+        # TODO AAM: let OTs manually change thresholds and deadband before timing tests
+        self._deadband = rec_deadband_size
+        self._sip_threshold = rec_sip_threshold
+        self._puff_threshold = rec_puff_threshold
+
+        print("Starting Ramp up/down timing tests...")
+        self._puff_ramp_times_test()
+        self._sip_ramp_times_test()
+
+        self._params_set = True
+
+    def _threshold_test(self):
+        # Takes 5 pressure readings when desired
+        pressures = []
+        for i in range(NUM_SAMPLES):
+            input("Press Enter to take measurement {}".format(i + 1))
+            # sleep to allow for data from read thread to come in
+            time.sleep(0.1)
+            pressure = self.getPressure()
+            pressures.append(pressure)
+        print(pressures)
+        next_state_prompt = input("Retry this state? (y/n)")
+        if next_state_prompt is 'y':
+            pressures = self._threshold_test()
+        
+        return pressures
+            
+    def _puff_ramp_times_test(self):
+        # tests puff ramp up/down times
+        # ramp up time starts when the pressure leaves the deadband
+        deadband_threshold = self._ambient_pressure+self._deadband
+        input("Press Enter when ready to begin")
+        time.sleep(0.1)
+        pressure = self.getPressure()
+        # loop blocks until pressure is outside deadband
+        print("Looking for hard puff...")
+        while pressure <= self._puff_threshold:
+            time.sleep(0.05)
+            pressure = self.getPressure()
+            # if pressure is within deadband, reset start time
+            if pressure < deadband_threshold:
+                ramp_up_start_time = time.time()
+            # print(pressure)
+        ramp_up_time = time.time() - ramp_up_start_time
+        while pressure >= deadband_threshold:
+            time.sleep(0.025)
+            pressure = self.getPressure()
+            # if pressure is within deadband, reset start time
+            if pressure > self._puff_threshold:
+                ramp_down_start_time = time.time()
+            # print(pressure)
+        ramp_down_time = time.time() - ramp_down_start_time
+
+        # TODO: Check if took more than max allowable time (2000 ms)
+        print("Hard puff ramped up in {}".format(ramp_up_time))
+        print("Hard puff ramped down in {}".format(ramp_down_time))
+
+        if input("Use these values? (y/n) ") is 'n':
```

Comment:
I think we will want to check if they just want to input values at the very beginning so that they don't have to go through the entire setup procedure before deciding to input their own values.